packages/stylify/src/Compiler/Compiler.ts

```diff
@@ -199,19 +199,40 @@ export class Compiler {
 		selector: string,
 		componentConfig: ComponentSelectorsType|ComponentConfigInterface
 	): Compiler {
-		if (selector in this.components) {
+		if (selector.includes(',')) {
+			selector.split(',').forEach((selector) => {
+				this.addComponent(selector.trim(), componentConfig);
+			});
+			return;
+		}
+
+		const selectorIsComponent = selector in this.components;
+
+		if (selectorIsComponent && this.components[selector].processed === true) {
+			const info = `You are trying to overwrite component ${selector} that has already been processed.`;
+			if (this.dev) {
+				console.warn(info);
+			} else {
+				throw new Error(info);
+			}
 			return;
 		}
 
 		if (typeof componentConfig === 'string' || Array.isArray(componentConfig)) {
 			componentConfig = {
-				selectors: componentConfig,
-				selectorsChain: []
+				selectors: this.convertStringOrStringArrayToFilteredArray([
+					...Array.isArray(componentConfig) ? componentConfig: [componentConfig],
```

Comment:
The result of this check `Array.isArray(componentConfig)` could be stored in a variable and used in the condition on line 221 and here.